js/mainjs/addIPCListeners.js

```diff
@@ -30,7 +34,10 @@ module.exports = function(config, mainWindow) {
 			default:
 				console.error('Unknown dialog ipc');
 		}
-		event.returnValue = response ? response : null;
+		// TODO: Make async and adapt plugins
```

Comment:
It's not that hard. Currently dialog windows don't block the UI, a minor annoyance but makes for slightly easier to read code. From the [dialog module in the electron documentation](http://electron.atom.io/docs/v0.36.0/api/dialog/):
>If a callback is passed, the API call will be asynchronous and the result will be passed via callback(filenames)